### ISSUE TITLE
Some issue exposed by LMDB

### ISSUE THREAD

**Issue by AlexDaniel** [rakudo/rakudo#2685]
See https://github.com/perl6/ecosystem-unbitrot/issues/445

Note that tests of this module don't pass. On older rakudo versions you get:
```
The following candidates are already installed: LMDB
===> Searching for: LMDB
===> Found: LMDB:ver<0.0.1>:auth<github:salortiz> [via Zef::Repository::LocalCache]
===> Testing: LMDB:ver<0.0.1>:auth<github:salortiz>
# To be continued...
t/01-basic.t .... ok
t/02-quick.t .... ok
# Failed test 'The ordered data'
# at t/03-custom.t line 34
# expected: $("AZ", IntStr.new(1, "1"), "BY", IntStr.new(2, "2"), "CX", IntStr.new(3, "3"), "DW", IntStr.new(4, "4"), "EV", IntStr.new(5, "5"))
#      got: $("AZ", "1", "BY", "2", "CX", "3", "DW", "4", "EV", "5")
# Looks like you failed 1 test of 19
t/03-custom.t ... 
Dubious, test returned 1 (wstat 256, 0x100)
Failed 1/19 subtests 
t/99.my-meta.t .. ok

Test Summary Report
-------------------
t/03-custom.t (Wstat: 256 Tests: 19 Failed: 1)
  Failed test:  8
  Non-zero exit status: 1
Files=4, Tests=109,  7 wallclock secs ( 0.04 usr  0.00 sys + 10.05 cusr  0.40 csys = 10.49 CPU)
Result: FAIL
===> Testing [FAIL]: LMDB:ver<0.0.1>:auth<github:salortiz>
Failed to get passing tests, but continuing with --force-test
```

Which is just a single innocent test with some sort of issue with IntStr. **That is fine.**

But on newer rakudos:
````````````````````````````````````````````````````````
===> Searching for: LMDB
===> Found: LMDB:ver<0.0.1>:auth<github:salortiz> [via Zef::Repository::Ecosystems<p6c>]
===> Fetching [OK]: LMDB:ver<0.0.1>:auth<github:salortiz> to /home/alex/Blin/data/zef-data/tmp/1549581459.8696.6792/p6-LMDB.git
===> Extraction [OK]: LMDB to /home/alex/Blin/data/zef-data/store/p6-LMDB.git
===> Testing: LMDB:ver<0.0.1>:auth<github:salortiz>
# To be continued...
t/01-basic.t .... ok
Cannot modify an immutable Str (Nil)
  in block <unit> at t/02-quick.t line 20

# Looks like you planned 14 tests, but ran 2
t/02-quick.t .... 
Dubious, test returned 255 (wstat 65280, 0xff00)
Failed 12/14 subtests 
Cannot modify an immutable Str (Nil)
  in block <unit> at t/03-custom.t line 33

# Looks like you planned 19 tests, but ran 7
t/03-custom.t ... 
Dubious, test returned 255 (wstat 65280, 0xff00)
Failed 12/19 subtests 
t/99.my-meta.t .. ok

Test Summary Report
-------------------
t/02-quick.t  (Wstat: 65280 Tests: 2 Failed: 0)
  Non-zero exit status: 255
  Parse errors: Bad plan.  You planned 14 tests but ran 2.
t/03-custom.t (Wstat: 65280 Tests: 7 Failed: 0)
  Non-zero exit status: 255
  Parse errors: Bad plan.  You planned 19 tests but ran 7.
Files=4, Tests=85, 15 wallclock secs ( 0.05 usr  0.01 sys + 12.68 cusr  0.44 csys = 13.18 CPU)
Result: FAIL
===> Testing [FAIL]: LMDB:ver<0.0.1>:auth<github:salortiz>
Failed to get passing tests, but continuing with --force-test
===> Installing: LMDB:ver<0.0.1>:auth<github:salortiz>
===> Install [OK] for LMDB:ver<0.0.1>:auth<github:salortiz>

````````````````````````````````````````````````````````

Or even:
````
The following candidates are already installed: LMDB
===> Searching for: LMDB
===> Found: LMDB:ver<0.0.1>:auth<github:salortiz> [via Zef::Repository::LocalCache]
===> Testing: LMDB:ver<0.0.1>:auth<github:salortiz>
Can only store CArray attribute in CArray slot in CStruct
  in submethod BUILD at /home/alex/git/Blin/data/zef-data/store/p6-LMDB.git/9edd0c83b22092f918f9635591eaccb5bf4dead7/lib/LMDB.pm6 (LMDB) line 96
  in method new-from-buf at /home/alex/git/Blin/data/zef-data/store/p6-LMDB.git/9edd0c83b22092f918f9635591eaccb5bf4dead7/lib/LMDB.pm6 (LMDB) line 99
  in method new-from-str at /home/alex/git/Blin/data/zef-data/store/p6-LMDB.git/9edd0c83b22092f918f9635591eaccb5bf4dead7/lib/LMDB.pm6 (LMDB) line 102
  in method put at /home/alex/git/Blin/data/zef-data/store/p6-LMDB.git/9edd0c83b22092f918f9635591eaccb5bf4dead7/lib/LMDB.pm6 (LMDB) line 433
  in block <unit> at t/01-basic.t line 80

# Looks like you planned 75 tests, but ran 36
t/01-basic.t .... 
Dubious, test returned 255 (wstat 65280, 0xff00)
Failed 39/75 subtests 
Can only store CArray attribute in CArray slot in CStruct
  in submethod BUILD at /home/alex/git/Blin/data/zef-data/store/p6-LMDB.git/9edd0c83b22092f918f9635591eaccb5bf4dead7/lib/LMDB.pm6 (LMDB) line 96
  in method get at /home/alex/git/Blin/data/zef-data/store/p6-LMDB.git/9edd0c83b22092f918f9635591eaccb5bf4dead7/lib/LMDB.pm6 (LMDB) line 445
  in method <anon> at /home/alex/git/Blin/data/zef-data/store/p6-LMDB.git/9edd0c83b22092f918f9635591eaccb5bf4dead7/lib/LMDB.pm6 (LMDB) line 573
  in method AT-KEY at /home/alex/git/Blin/data/zef-data/store/p6-LMDB.git/9edd0c83b22092f918f9635591eaccb5bf4dead7/lib/LMDB.pm6 (LMDB) line 571
  in block <unit> at t/02-quick.t line 20

# Looks like you planned 14 tests, but ran 2
t/02-quick.t .... 
Dubious, test returned 255 (wstat 65280, 0xff00)
Failed 12/14 subtests 
Can only store CArray attribute in CArray slot in CStruct
  in submethod BUILD at /home/alex/git/Blin/data/zef-data/store/p6-LMDB.git/9edd0c83b22092f918f9635591eaccb5bf4dead7/lib/LMDB.pm6 (LMDB) line 96
  in method get at /home/alex/git/Blin/data/zef-data/store/p6-LMDB.git/9edd0c83b22092f918f9635591eaccb5bf4dead7/lib/LMDB.pm6 (LMDB) line 445
  in method <anon> at /home/alex/git/Blin/data/zef-data/store/p6-LMDB.git/9edd0c83b22092f918f9635591eaccb5bf4dead7/lib/LMDB.pm6 (LMDB) line 573
  in method AT-KEY at /home/alex/git/Blin/data/zef-data/store/p6-LMDB.git/9edd0c83b22092f918f9635591eaccb5bf4dead7/lib/LMDB.pm6 (LMDB) line 571
  in block <unit> at t/03-custom.t line 33

# Looks like you planned 19 tests, but ran 7
t/03-custom.t ... 
Dubious, test returned 255 (wstat 65280, 0xff00)
Failed 12/19 subtests 
t/99.my-meta.t .. ok

Test Summary Report
-------------------
t/01-basic.t  (Wstat: 65280 Tests: 36 Failed: 0)
  Non-zero exit status: 255
  Parse errors: Bad plan.  You planned 75 tests but ran 36.
t/02-quick.t  (Wstat: 65280 Tests: 2 Failed: 0)
  Non-zero exit status: 255
  Parse errors: Bad plan.  You planned 14 tests but ran 2.
t/03-custom.t (Wstat: 65280 Tests: 7 Failed: 0)
  Non-zero exit status: 255
  Parse errors: Bad plan.  You planned 19 tests but ran 7.
Files=4, Tests=46,  3 wallclock secs ( 0.02 usr  0.02 sys +  3.25 cusr  0.20 csys =  3.49 CPU)
Result: FAIL
===> Testing [FAIL]: LMDB:ver<0.0.1>:auth<github:salortiz>
Failed to get passing tests, but continuing with --force-test
===> Installing: LMDB:ver<0.0.1>:auth<github:salortiz>
===> Install [OK] for LMDB:ver<0.0.1>:auth<github:salortiz>
````

Don't know what's going on.

**Comment by jonathanstowe** [comment]
This appears to be similar to #2687  on the face of it.

**Comment by AlexDaniel** [comment]
See https://github.com/rakudo/rakudo/issues/2681

**Comment by lizmat** [comment]
Fixed with 5e5c909d1a

**Comment by lizmat** [comment]
Duplicate of #2687
